lib/utils/docker-coffee.coffee

```diff
@@ -114,12 +113,12 @@ generateConnectOpts = (opts) ->
 		else if opts.docker? and opts.dockerHost?
 			# Both provided, no obvious way to continue
 			throw new Error("Both a local docker socket and docker host have been provided. Don't know how to continue.")
-		else if process.env.DOCKER_HOST
-			# If no explicit options are provided, use the env
-			connectOpts = buildDockerodeOpts(process.env.DOCKER_HOST)
 		else
-			# No options anywhere, assume default docker local socket
-			connectOpts.socketPath = '/var/run/docker.sock'
+			# Use docker-modem defaults which take the DOCKER_HOST env var into account
+			# https://github.com/apocas/docker-modem/blob/v2.0.2/lib/modem.js#L16-L65
+			Modem = require('docker-modem')
+			defaultOpts = new Modem()
+			connectOpts[opt] = defaultOpts[opt] for opt in ['host', 'port', 'socketPath']
```

Comment:
These changes fix two related issues:
* #1132, where the default socketPath value on Windows should be `//./pipe/docker_engine` instead of `/var/run/docker.sock`.
* https://github.com/balena-io/balena-cli/issues/1132#issuecomment-532422385 When the DOCKER_HOST environment variable is set to a value such as `unix:///run/user/1000/docker.sock`, the previous implementation using `require('dockerode-options')` was unable to extract the correct path. I found that the `docker-modem` v2 package [supports `unix://` and `npipe://` URLs](https://github.com/apocas/docker-modem/blob/v2.0.2/lib/modem.js#L16-L65) in the DOCKER_HOST env var, so I've changed the implementation to use it instead.